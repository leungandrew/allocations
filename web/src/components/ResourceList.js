import React from 'react';
import Resource from './Resource';

const ResourceList = ({resources}) => {
  return (
    <div>
      {
        resources.map( resource => {
          return <Resource id={resource.id} key={resource.id} resource={resource} />
        })
      }
    </div>
  )
};

export default ResourceList;
