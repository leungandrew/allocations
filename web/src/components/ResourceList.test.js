import React from 'react';
import ResourceList from './ResourceList';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

let resources;

describe('ResourceList', () => {

  beforeEach(() => {
    resources = [
      {
        id: 'id-1',
        firstName: 'FirstUser1',
        lastName: 'LastUser1'
      },
      {
        id: 'id-2',
        firstName: 'FirstUser2',
        lastName: 'LastUser2'
      }
    ];
  });

  it('renders a list of Resources', () => {
    const subject = shallow(<ResourceList resources={resources} />);
    expect(toJson(subject)).toMatchSnapshot();
  });
});

