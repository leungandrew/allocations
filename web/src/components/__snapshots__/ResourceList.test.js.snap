// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceList renders a list of Resources 1`] = `
<div>
  <Resource
    id="id-1"
    key="id-1"
    resource={
      Object {
        "firstName": "FirstUser1",
        "id": "id-1",
        "lastName": "LastUser1",
      }
    }
  />
  <Resource
    id="id-2"
    key="id-2"
    resource={
      Object {
        "firstName": "FirstUser2",
        "id": "id-2",
        "lastName": "LastUser2",
      }
    }
  />
</div>
`;
